pkg/controller/replication/conversion.go

```diff
@@ -0,0 +1,307 @@
+}
+
+func (l conversionLister) List(selector labels.Selector) ([]*extensionsv1beta1.ReplicaSet, error) {
+	rcList, err := l.rcLister.List(selector)
+	if err != nil {
+		return nil, err
+	}
+	return convertSlice(rcList)
+}
+
+func (l conversionLister) ReplicaSets(namespace string) extensionslisters.ReplicaSetNamespaceLister {
+	return conversionNamespaceLister{l.rcLister.ReplicationControllers(namespace)}
+}
+
+func (l conversionLister) GetPodReplicaSets(pod *v1.Pod) ([]*extensionsv1beta1.ReplicaSet, error) {
+	rcList, err := l.rcLister.GetPodControllers(pod)
+	if err != nil {
+		return nil, err
+	}
+	return convertSlice(rcList)
+}
+
+type conversionNamespaceLister struct {
+	rcLister v1listers.ReplicationControllerNamespaceLister
+}
+
+func (l conversionNamespaceLister) List(selector labels.Selector) ([]*extensionsv1beta1.ReplicaSet, error) {
+	rcList, err := l.rcLister.List(selector)
+	if err != nil {
+		return nil, err
+	}
+	return convertSlice(rcList)
+}
+
+func (l conversionNamespaceLister) Get(name string) (*extensionsv1beta1.ReplicaSet, error) {
+	rc, err := l.rcLister.Get(name)
+	if err != nil {
+		return nil, err
+	}
+	return convertRCtoRS(rc, nil)
+}
+
+type conversionEventHandler struct {
+	handler cache.ResourceEventHandler
+}
+
+func (h conversionEventHandler) OnAdd(obj interface{}) {
+	if rs, err := convertRCtoRS(obj.(*v1.ReplicationController), nil); err == nil {
+		obj = rs
+	}
+	h.handler.OnAdd(obj)
+}
+
+func (h conversionEventHandler) OnUpdate(oldObj, newObj interface{}) {
+	if oldRS, err := convertRCtoRS(oldObj.(*v1.ReplicationController), nil); err == nil {
+		oldObj = oldRS
+	}
+	if newRS, err := convertRCtoRS(newObj.(*v1.ReplicationController), nil); err == nil {
+		newObj = newRS
+	}
+	h.handler.OnUpdate(oldObj, newObj)
+}
+
+func (h conversionEventHandler) OnDelete(obj interface{}) {
+	var rs extensionsv1beta1.ReplicaSet
+	if legacyscheme.Scheme.Convert(obj, &rs, nil) == nil {
+		obj = &rs
+	}
+	h.handler.OnDelete(obj)
+}
+
+type clientsetAdapter struct {
+	clientset.Interface
+}
+
+func (c clientsetAdapter) ExtensionsV1beta1() extensionsv1beta1client.ExtensionsV1beta1Interface {
+	return conversionExtensionsClient{c.Interface, c.Interface.ExtensionsV1beta1()}
+}
+
+func (c clientsetAdapter) Extensions() extensionsv1beta1client.ExtensionsV1beta1Interface {
+	return conversionExtensionsClient{c.Interface, c.Interface.Extensions()}
+}
+
+type conversionExtensionsClient struct {
+	clientset clientset.Interface
+	extensionsv1beta1client.ExtensionsV1beta1Interface
+}
+
+func (c conversionExtensionsClient) ReplicaSets(namespace string) extensionsv1beta1client.ReplicaSetInterface {
+	return conversionClient{c.clientset.CoreV1().ReplicationControllers(namespace)}
+}
+
+type conversionClient struct {
+	v1client.ReplicationControllerInterface
+}
+
+func (c conversionClient) Create(rs *extensionsv1beta1.ReplicaSet) (*extensionsv1beta1.ReplicaSet, error) {
+	return convertPut(rs, c.ReplicationControllerInterface.Create)
```

Comment:
nit: create uses POST not PUT

Reply:
It's probably not too much to add a separate convertPost, but it would duplicate the converPut method exactly. Are you thinking something like convertPuyOrPost?